nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -193,6 +231,74 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
 
     }
 
+    private Statement generateUpdate(String cassandraTable, RecordSchema schema, String updateKeys, String updateMethod, Map<String, Object> recordContentMap) {
+        Update updateQuery;
+
+        // Split up the update key names separated by a comma, we need at least 1 key.
+        final Set<String> updateKeyNames;
+        updateKeyNames = new HashSet<>();
+        for (final String updateKey : updateKeys.split(",")) {
+            updateKeyNames.add(updateKey.trim());
```

Comment:
You may want to check for the empty string here before adding to the set. There's a common pattern you see in NiFi, something like:

```
updateKeyNames = Arrays.stream(updateKeys.split(","))
   .map(String::trim)
   .filter(StringUtils::isNotEmpty)
   collect(Collectors.toSet());
```